src/solidlsp/language_servers/scala_language_server.py

```diff
@@ -0,0 +1,267 @@
+"""
+Provides Scala specific instantiation of the LanguageServer class. Contains various configurations and settings specific to Scala.
+"""
+
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import threading
+
+from overrides import override
+
+from solidlsp.ls import SolidLanguageServer
+from solidlsp.ls_config import LanguageServerConfig
+from solidlsp.ls_logger import LanguageServerLogger
+from solidlsp.ls_utils import PlatformUtils
+from solidlsp.lsp_protocol_handler.lsp_types import InitializeParams
+from solidlsp.lsp_protocol_handler.server import ProcessLaunchInfo
+from solidlsp.settings import SolidLSPSettings
+
+# Platform-specific imports
+if os.name != "nt":  # Unix-like systems
+    import pwd
+else:
+    # Dummy pwd module for Windows
+    class pwd:
+        @staticmethod
+        def getpwuid(uid):
+            return type("obj", (), {"pw_name": os.environ.get("USERNAME", "unknown")})()
+
+
+if not PlatformUtils.get_platform_id().value.startswith("win"):
+    pass
+
+
+class ScalaLanguageServer(SolidLanguageServer):
+    """
+    Provides Scala specific instantiation of the LanguageServer class. Contains various configurations and settings specific to Scala.
+    """
+
+    def __init__(
+        self, config: LanguageServerConfig, logger: LanguageServerLogger, repository_root_path: str, solidlsp_settings: SolidLSPSettings
+    ):
+        """
+        Creates a ScalaLanguageServer instance. This class is not meant to be instantiated directly. Use LanguageServer.create() instead.
+        """
+        scala_lsp_executable_path = self._setup_runtime_dependencies(logger, config, solidlsp_settings)
+        super().__init__(
+            config,
+            logger,
+            repository_root_path,
+            ProcessLaunchInfo(cmd=scala_lsp_executable_path, cwd=repository_root_path),
+            config.code_language.value,
+            solidlsp_settings,
+        )
+        self.server_ready = threading.Event()
+        self.initialize_searcher_command_available = threading.Event()
+
+    @override
+    def is_ignored_dirname(self, dirname: str) -> bool:
+        return super().is_ignored_dirname(dirname) or dirname in [
+            ".bloop",
+            ".metals",
+            "target",
+        ]
+
+    @classmethod
+    def _setup_runtime_dependencies(
+        cls, logger: LanguageServerLogger, config: LanguageServerConfig, solidlsp_settings: SolidLSPSettings
+    ) -> list[str]:
+        """
+        Setup runtime dependencies for Scala Language Server and return the command to start the server.
+        """
+        assert shutil.which("java") is not None, "JDK is not installed or not in PATH."
+
+        metals_home = os.path.join(cls.ls_resources_dir(solidlsp_settings), "metals-lsp")
+        os.makedirs(metals_home, exist_ok=True)
+        metals_executable = os.path.join(metals_home, "metals")
+        coursier_command_path = shutil.which("coursier")
```

Comment:
you only need coursier to install metals, right? Then all the checks should only be performed if metals_executable was not found